streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala

```diff
@@ -308,12 +336,25 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
         supervisor.start()
         supervisor.awaitTermination()
       }
+
       // Run the dummy Spark job to ensure that all slaves have registered.
       // This avoids all the receivers to be scheduled on the same node.
       if (!ssc.sparkContext.isLocal) {
         ssc.sparkContext.makeRDD(1 to 50, 50).map(x => (x, 1)).reduceByKey(_ + _, 20).collect()
       }
 
+      // Get the list of executors and schedule receivers
+      val executors = getExecutors(ssc)
+      val locations = scheduleReceivers(receivers, executors)
+      val tempRDD =
+        if (locations(0) != null) {
```

Comment:
That's fine. Following, I think, is slightly cleaner:

if(!executors.isEmpty){
   scheduleReceivers
   make RDD with returned result
}else{
  make RDD
}
Avoids the redundant invocation to scheduleReceivers and subsequent memory allocations. Optimizes away the extra logic (check on local) and assumptions about how locations is formatted. If this sounds good, I think we can get the final iteration of this PR going.